src/Nancy/Routing/Trie/RouteResolverTrie.cs

```diff
@@ -62,19 +62,15 @@ public void BuildTrie(IRouteCache cache)
         /// <returns>An array of <see cref="MatchResult"/> elements</returns>
         public MatchResult[] GetMatches(string method, string path, NancyContext context)
         {
-            if (string.IsNullOrEmpty(path))
-            {
-                return MatchResult.NoMatches;
-            }
+            TrieNode result;
 
-            // TODO -concurrent if allowing updates?
-            if (!this.routeTries.ContainsKey(method))
+            if (!this.routeTries.TryGetValue(method, out result))
             {
                 return MatchResult.NoMatches;
             }
 
-            return this.routeTries[method].GetMatches(path.Split(splitSeparators, StringSplitOptions.RemoveEmptyEntries), context)
-                                          .ToArray();
+            return result.GetMatches(path.Split(splitSeparators, StringSplitOptions.RemoveEmptyEntries), context)

```

Comment:
Leave it out
On Feb 27, 2016 1:37 PM, "Phillip Haydon" <notifications@github.com> wrote:
>
> In src/Nancy/Routing/Trie/RouteResolverTrie.cs:
>
> >              {
> >                  return MatchResult.NoMatches;
> >              }
> >
> > -            return
this.routeTries[method].GetMatches(path.Split(splitSeparators,
StringSplitOptions.RemoveEmptyEntries), context)
> > -                                          .ToArray();
> > +            return result.GetMatches(path.Split(splitSeparators,
StringSplitOptions.RemoveEmptyEntries), context)
>
> HTTP Spec states a path must start with "/" is be empty. If a web server
does not conform to this then the request is invalid.
>
> If you ABSOLUTELY want me to put the check in there. I can. It's not
expensive to do the null check. It's just unnecessary.
>
> —
> Reply to this email directly or view it on GitHub.